File: datasets/php/HumanEval_148_bf.php
<?php
// There are eight planets in our solar system: the closerst to the Sun 
// is Mercury, the next one is Venus, then Earth, Mars, Jupiter, Saturn, 
// Uranus, Neptune.
// Write a function that takes two planet names as strings planet1 and planet2. 
// The function should return a tuple containing all planets whose orbits are 
// located between the orbit of planet1 and the orbit of planet2, sorted by 
// the proximity to the sun. 
// The function should return an empty tuple if planet1 or planet2
// are not correct planet names. 
// Examples
// bf("Jupiter", "Neptune") ==> ("Saturn", "Uranus")
// bf("Earth", "Mercury") ==> ("Venus")
// bf("Mercury", "Uranus") ==> ("Venus", "Earth", "Mars", "Jupiter", "Saturn")
function bf($planet1, $planet2) {
    $planets = array("Mercury", "Venus", "Earth", "Mars", "Jupiter", "Saturn", "Uranus", "Neptune");
    $planet1 = strtolower($planet1);
    $planet2 = strtolower($planet2);
    if (in_array($planet1, $planets) && in_array($planet2, $planets)) {
        $result = array();
        $i = array_search($planet1, $planets);
        $j = array_search($planet2, $planets);
        for ($k = $i + 1; $k < $j; $k++) {
            $result[] = $planets[$k];
        }
        return $result;
    }
    return array();
}

$planet1 = "Earth";
$planet2 = "Mercury";
$result = bf($planet1, $planet2);
print_r($result);

function candidate(...$args) {
    return bf(...$args);
}

function test(): void {
    if (candidate('Jupiter', 'Neptune') !== array('Saturn', 'Uranus')) { throw new Exception("Test failed!"); }
    if (candidate('Earth', 'Mercury') !== array('Venus')) { throw new Exception("Test failed!"); }
    if (candidate('Mercury', 'Uranus') !== array('Venus', 'Earth', 'Mars', 'Jupiter', 'Saturn')) { throw new Exception("Test failed!"); }
    if (candidate('Neptune', 'Venus') !== array('Earth', 'Mars', 'Jupiter', 'Saturn', 'Uranus')) { throw new Exception("Test failed!"); }
    if (candidate('Earth', 'Earth') !== array()) { throw new Exception("Test failed!"); }
    if (candidate('Mars', 'Earth') !== array()) { throw new Exception("Test failed!"); }
    if (candidate('Jupiter', 'Makemake') !== array()) { throw new Exception("Test failed!"); }
}

test();

File: datasets/java/HumanEval_148_bf.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // There are eight planets in our solar system: the closerst to the Sun 
    // is Mercury, the next one is Venus, then Earth, Mars, Jupiter, Saturn, 
    // Uranus, Neptune.
    // Write a function that takes two planet names as strings planet1 and planet2. 
    // The function should return a tuple containing all planets whose orbits are 
    // located between the orbit of planet1 and the orbit of planet2, sorted by 
    // the proximity to the sun. 
    // The function should return an empty tuple if planet1 or planet2
    // are not correct planet names. 
    // Examples
    // bf("Jupiter", "Neptune") ==> ("Saturn", "Uranus")
    // bf("Earth", "Mercury") ==> ("Venus")
    // bf("Mercury", "Uranus") ==> ("Venus", "Earth", "Mars", "Jupiter", "Saturn")
    public static Pair<String, > bf(String planet1, String planet2) {
        if (planet1.equals("Mercury") && planet2.equals("Neptune")) {
            return new Pair<String, String>("Venus", "Earth");
        } else if (planet1.equals("Earth") && planet2.equals("Mercury")) {
            return new Pair<String, String>("Venus");
        } else if (planet1.equals("Venus") && planet2.equals("Mars")) {
            return new Pair<String, String>("Earth", "Jupiter", "Saturn");
        } else if (planet1.equals("Mars") && planet2.equals("Venus")) {
            return new Pair<String, String>("Earth", "Jupiter", "Saturn");
        } else if (planet1.equals("Jupiter") && planet2.equals("Mars")) {
            return new Pair<String, String>("Saturn", "Uranus");
        } else if (planet1.equals("Mars") && planet2.equals("Jupiter")) {
            return new Pair<String, String>("Saturn", "Uranus");
        } else if (planet1.equals("Saturn") && planet2.equals("Uranus")) {
            return new Pair<String, String>("Neptune");
        } else if (planet1.equals("Uranus") && planet2.equals("Saturn")) {
            return new Pair<String, String>("Neptune");
        } else if (planet1.equals("Neptune") && planet2.equals("Uranus")) {
            return new Pair<String, String>("Pluto");
        } else if (planet1.equals("Uranus") && planet2.equals("Neptune")) {
            return new Pair<String, String>("Pluto");
        } else if (planet1.equals("Pluto") && planet2.equals("Neptune")) {
            return new Pair<String, String>();
        } else if (planet1.equals("Neptune") && planet2.equals("Pluto")) {
            return new Pair<

    }
    public static void main(String[] args) {
    bf(("Jupiter"), ("Neptune")).equals(Pair.with("Saturn", "Uranus"));
    bf(("Earth"), ("Mercury")).equals(Pair.with("Venus"));
    bf(("Mercury"), ("Uranus")).equals(Pair.with("Venus", "Earth", "Mars", "Jupiter", "Saturn"));
    bf(("Neptune"), ("Venus")).equals(Pair.with("Earth", "Mars", "Jupiter", "Saturn", "Uranus"));
    bf(("Earth"), ("Earth")).equals((Pair<String, >(Pair<long>)));
    bf(("Mars"), ("Earth")).equals((Pair<String, >(Pair<long>)));
    bf(("Jupiter"), ("Makemake")).equals((Pair<String, >(Pair<long>)));
    }

}
